pam_tcb/Makefile

```diff
@@ -4,6 +4,10 @@ PAM_SO_SUFFIX =
 PAM_TCB = pam_tcb.so$(PAM_SO_SUFFIX)
 PAM_MAP = pam_tcb.map
 
+ifneq ($(PAM_SO_SUFFIX),)
+PAM_TCB_SONAME = -Wl,-soname,$(PAM_TCB)
+endif
```

Comment:
Why do you conditionalize on `PAM_SO_SUFFIX`?
If you say every shared library should have a soname, then `$(PAM_TCB)` is either a shared library or not regardless of `PAM_SO_SUFFIX`.
I'm not sure pam_tcb is a shared library, though.  Yes, it's a shared object, but it looks more like a plugin to me.
You don't link with it, do you?

Reply:
No, I don't but, OpenPAM expects its modules to have a soname matching the module name + `PAM_SO_SUFFIX`.  With PAM-Linux we don't  have either to deal with.